plugins/logs/plugin.go

```diff
@@ -771,13 +781,18 @@ func (p *Plugin) maskEvent(ctx context.Context, txn storage.Transaction, event *
 	return nil
 }
 
-func uploadChunk(ctx context.Context, client rest.Client, partitionName string, data []byte) error {
+func uploadChunk(ctx context.Context, client rest.Client, resourcePath string, partitionName string, data []byte) error {
+
+	logPath := fmt.Sprintf("%v", resourcePath)
+	if partitionName != "" {
+		logPath = fmt.Sprintf("/logs/%v", partitionName)
```

Comment:
No resourcePath here? 💭

Reply:
Nevermind I've misunderstood the intention here, I think. This path is basically how it worked before. Hmm maybe we should clarify that in the docs? I'm not sure how uncommon my (wrong) expectation here is.

Reply:
yep that is how partition name worked previously, by appending it to `/logs/<partition>`. I wasn't totally sure how this should be handled, but my thought was to remain backward compatible (for now) we make the partition name take precedence over resource. I think it's a good idea to make note of that in the docs though! I'll push a change to add some info to the docs about this, thanks!

Reply:
I added some more docs in cea605a, let me know if that wording/format makes sense or if there's more detail I should add!

Reply:
I'd recommend moving this logic into the `validateAndInjectDefaults` function or onto the Config struct as a helper function. That way this code can remain more or less the same and we can test the validation/defaulting behaviour as a smaller unit.

As far as the precedence goes, this looks good to me...

1. If partitionName is set use /logs/<partitionName>
2. If resource is set use <resource>
3. Otherwise use /logs

The resource and partitionName fields should be mutually exclusive. We should be able to implement all of this during validation/defaulting.

Reply:
moved to validateAndInjectDefaults in d5fd809